cpp/modmesh/buffer/SimpleArray.cpp

```diff
@@ -36,6 +36,77 @@ namespace modmesh
 
 namespace detail
 {
+
+#if defined(__aarch64__)
+#define vec_typ(typ, N) typ##x##N##_t
+#define t_typ(typ) typ##_t
+
+#define IMPL_CHECK_IDX_RNG_SIMD(typ, N_vec, typ_symbol)                                                       \
+    template <>                                                                                               \
+    t_typ(typ) const * check_index_range<t_typ(typ)>(SimpleArray<t_typ(typ)> const & indices, size_t max_idx) \
+    {                                                                                                         \
+        check_type_range(t_typ(typ), max_idx);                                                                \
+                                                                                                              \
+        vec_typ(typ, N_vec) max_vec = vdupq_n_##typ_symbol(static_cast<t_typ(typ)>(max_idx));                 \
+        vec_typ(typ, N_vec) data_vec = {};                                                                    \
+        vec_typ(typ, N_vec) cmp_vec = {};                                                                     \
+                                                                                                              \
+        t_typ(typ) const * src = indices.begin();                                                             \
+        t_typ(typ) const * const end = indices.end();                                                         \
+                                                                                                              \
+        for (; src <= end - N_vec; src += N_vec)                                                              \
+        {                                                                                                     \
+            data_vec = vld1q_##typ_symbol(src);                                                               \
+            cmp_vec = vcgeq_##typ_symbol(data_vec, max_vec);                                                  \
+            if (vgetq_lane_##typ_symbol(cmp_vec, 0) ||                                                        \
+                vgetq_lane_##typ_symbol(cmp_vec, 1))                                                          \
+            {                                                                                                 \
+                goto OUT_OF_RANGE;                                                                            \
```

Comment:
@yungyuc  do you accept this?

Reply:
GOTO in asm is OK.  But I need this to be moved in separate files before get a right feeling for the code structure. In short, I don't know yet.